### PR TITLE
chore: add key dict

### PR DESCRIPTION
**Pull request by azanbinzahid**
Related PR: https://github.com/edx/edx-internal/pull/7137

### CODE REVIEW

File: course_discovery/settings/production.py [hunk 1]
```diff
@@ -89,21 +89,5 @@
...
-    GOOGLE_SERVICE_ACCOUNT_CREDENTIALS = json.loads(GOOGLE_SERVICE_ACCOUNT_CREDENTIALS)
-except json.decoder.JSONDecodeError:
-    logger.warning("GOOGLE_SERVICE_ACCOUNT_CREDENTIALS were not in expected format")
-except Exception: # pylint: disable=broad-except
-    logger.warning(
-        "Cannot decode GOOGLE_SERVICE_ACCOUNT_CREDENTIALS value of type: %s",
-        type(GOOGLE_SERVICE_ACCOUNT_CREDENTIALS),
-    )
+# Convert dict keys to lowercase
+GOOGLE_SERVICE_ACCOUNT_CREDENTIALS = {k.lower(): v for k, v in GOOGLE_SERVICE_ACCOUNT_CREDENTIALS.iteritems()}
```
Comment (DawoudSheraz) [inline course_discovery/settings/production.py:93]: do we need to add it in DICT_UPDATE_KEYS settings?

Comment (azanbinzahid) [reply course_discovery/settings/production.py:93]: I believe no. I am following the example of `ALGOLIA`

Comment (DawoudSheraz) [reply course_discovery/settings/production.py:93]: Link/reference? I can't seem to find it in production.py.

Comment (azanbinzahid) [reply course_discovery/settings/production.py:93]: https://github.com/edx/edx-internal/blob/455c7c4885d6923a77ae21172e46e85dc271e0d7/edx-remote-config/stage/discovery.yml#L203
https://github.com/openedx/course-discovery/blob/ecb9c9c17083049f8d9d3ec7a9127a3d64cc09ff/course_discovery/settings/base.py#L71

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**